admin-dev/themes/new-theme/js/pages/product/components/specific-price/specific-price-list-renderer.ts

```diff
@@ -102,23 +105,23 @@ export default class SpecificPriceListRenderer implements RendererType {
       editBtn.dataset.specificPriceId = String(specificPrice.id);
 
       if (!specificPrice.period) {
-        periodField.textContent = String(periodField.dataset.unlimitedText);
+        periodField.textContent = String(periodField?.dataset.unlimitedText);
       } else {
         periodFromField.textContent = specificPrice.period.from;
         periodToField.textContent = specificPrice.period.to;
       }
 
-      tbody.append(trClone);
+      tbody?.append(trClone);
       this.addEventListenerForDeleteBtn(deleteBtn);
     });
   }
 
   private toggleListVisibility(show: boolean): void {
-    this.listContainer.classList.toggle('d-none', !show);
+    this.listContainer?.classList.toggle('d-none', !show);
   }
 
-  private selectListField(templateTrClone: HTMLElement, selector: string): HTMLElement {
-    return templateTrClone.querySelector(selector) as HTMLElement;
+  private selectListField(templateTrClone: HTMLElement, selector: string): HTMLElement | null | undefined {
+    return templateTrClone.querySelector<HTMLElement>(selector);
```

Comment:
I'm not a big fan, but you can shortcircuit the type check by using 

```suggestion
    return templateTrClone.querySelector<HTMLElement>(selector)!;
```

Which doesn't change anything between the casting and the use of the `querySelector` generic and this

Checking every selected element would ensure that the app will work as expected but in that particular case this is really verbose and probably show an wrong implementation :/

Reply:
Thanks @NeOMakinG!

Doing shortcircuit to the return statement still got errors to the const assignment(s) :/
But doing to each assignment, i.e.
`const idField = this.selectListField(trClone, listFields.specificPriceId)!;`
(see new commit)

"works", but I don't think is the right way ^^" and, as you said, doesn't change anything, so we're back to square.

What do you suggest?

Reply:
I would suggest to keep the cast there which result the same but is less verbose

Reply:
Thanks!